src/goto-programs/goto_contractor.cpp

```diff
@@ -69,26 +72,25 @@ void goto_contractort::parse_intervals(expr2tc expr)
     neg = true;
     side2 = to_neg2t(side2).value;
   }
+
   if(!is_constant_int2t(side2))
     return;
 
-  value = to_constant_int2t(side2).as_long() * (neg ? -1 : 1);
+  value = to_constant_int2t(side2).as_long() * (neg ? -1 : 1) + (neg ? -1 : 1);
```

Comment:
What does this do exactly?

Reply:
When I use the `--interval-analysis` option it will insert assumes into the code. These assumes do not hold in some cases because of a difference of 1. For example:
```
int x=0;
int y=0;
while( y < 1000 )
{
   assume(y<=999) //inserted by interval analysis option
   x=x+y;
   y++;
}
assert(x >=y)
```
Here the interval analysis option says the upper bound is 999. However, when we reach the assert it will be 1000. that's why I added/subtracted 1 in that statement. It improved the results for some benchmarks.

I am currently working on a script that will extract accurate intervals using Frama-C.   This is just temporary.

Reply:
I see... The interval analysis is missing the loop termination condition that it's added but the k-induction transformations.

You can probably move that from the k-induction algorithm to a separate transformation step in a separate PR, so you can remove this code. WDYT @lucasccordeiro?

Reply:
Yes, I agree that it could be a separate PR.